### ISSUE TITLE
Street view camera

### ISSUE THREAD

**Issue by clement-igonet** [maplibre/maplibre-gl-js#4428]
## User Story

As a walker in the street, or at a mountain view point, I'd like to be able to move the view around a fixed GPS coordinate so that I can navigate in an immersive way.

## Rationale

Nowadays, more and more "photosphere" or "equirectangle" pictures are openly usable from web API, and it would be a killer feature to facilitate their use in maplibre library.

A P.O.C. had already been implemented as a "FPS" (First Person Shooter) extension at the old time of "OpenIndoor" project, extending ui/camera class, and move handlers. I suggest to include this feature natively in maplibre code now.

**Comment by HarelM** [comment]
Can this code be added as a plugin as an initial phase and see how widely this is used?
Even using an example page with this plugin in the docs to improve traction...?

**Comment by clement-igonet** [comment]
Of course, good point to work on adoption of the feature before implementing it natively.

May I use the current issue as a reference for coming work on the topic ?

**Comment by HarelM** [comment]
Sure, whatever works for you.

**Comment by clement-igonet** [comment]
OK, a maplibre/maplibre-gl-viewpoint github empty project would be welcome.

**Comment by HarelM** [comment]
Sorry, I misread what you wrote, please create it under your space.

**Comment by birkskyum** [comment]
Could be related to this request for controlling our camera:
- https://github.com/visgl/deck.gl/issues/9029#issuecomment-2231718621

**Comment by kfern** [comment]
Hi. Any progress?
